### DISCUSSION TITLE
User Group

### DISCUSSION THREAD

**Discussion by GuilhermeHirsch** [Budibase/budibase#2185] [Ideas]
Guys, a question arose about the user group. I'm an accountant, and my idea with budibase is to create a platform for each of my clients (small businesses). The user group would be very interesting for me, that is, my client cannot see or have access to other apps. This functionality will be paid for :( ? I only need this division, without much management complexity.

**Comment by mjashanks** [comment]
Hey @GuilhermeHirsch .

Budibase groups (due later this year) will definitely be a paid feature. You've made a lot of contribution to our community, so I'm sure we can work out a good deal for you when the time comes :)

**Reply by GuilhermeHirsch** [reply]
Wow! Thank you very much, @mjashanks! Great to hear that I'm collaborating in some way. It is a pleasure to collaborate with this tool. I've used many tools with a similar proposal, but Budibase is special, the vision, the way it was structured, the people behind it. I'm an accountant and I work with this, but I've been studying programming for 6 years (JavaScript, Python, Dart, Flutter...) in order to develop something for my accounting business. I found in no-code and low-code the possibility to develop something quickly and in a very professional, scalable way. Anyway, I just have to thank you.

**Comment by Rory-Powell** [comment]
Hey @GuilhermeHirsch 👋  we have two streams of work in upcoming releases that can help here, as mike mentioned they will be paid for features.

multi tenancy 
- data is segregated per tenant
- each tenant has a unique portal login
- users, smtp configs, sso configs etc are stored and accessed separately
- apps are scoped to a tenant

groups / workspaces
- all the above mentioned data is shared
- apps can be scoped to a group to limit access

**Reply by GuilhermeHirsch** [reply]
This looks pretty good, @Rory-Powell! That's exactly what I need. Thank you very much for the detail.

**Comment by KernOil** [comment]
Hi Team, @mjashanks @Rory-Powell 
I have a clarifying question, as of now there is no way to separate users so they can only see their app in their app portal, is that correct? currently, all BB users see all BB published apps is that correct, or am I missing something. The way to organize users and apps is in groups and that feature will come later and is paid?

**Reply by mjashanks** [reply]
Good point Andrew. I believe that when selecting a role for a user, we need an option for `No Access` - i.e. the user cannot see the app or access it anyway.

At this point, all users can list all apps, even if the `BASIC` role cannot actually do anything.

Even if we had a  `No Access` option, you would still need to administer this **per user**. Groups will provide a way to give a user all the access that they need by adding them to one or more groups.

**Reply by kevmodrome** [reply]
At the moment all users can see all apps that are published. You can change their access levels (per user) to only allow some of them the ability to access data. They will still be able to see the app in the dashboard though.

It's correct that in the future you will be able to split users into different groups and change access on a more global level (per group). This will be, as mentioned, a paid feature.

**Reply by kevmodrome** [reply]
A `No access` role sounds like a great idea.

**Reply by KernOil** [reply]
I think a no-access user role would work great. I have no problem going in and changing on a user-by-user basis. but I do think some separation is needed.

**Reply by GuilhermeHirsch** [reply]
In my case it would be perfect!

**Reply by GuilhermeHirsch** [reply]
Guys, any plans for implementing the "no access" rule? This would be essential in my business model.

**Reply by joebudi** [reply]
Hey @GuilhermeHirsch

We will try to get this implemented this month or next. Thanks!

**Reply by GuilhermeHirsch** [reply]
Wow! That's great! Thanks, Joe! Have a great day!

**Reply by joebudi** [reply]
Thanks @GuilhermeHirsch . You too!

**Reply by tobehn** [reply]
@joebudi and @mjashanks - Will these paid features be available for self-hosters? Do you plan to make them available as plugins? The User Group feature will become a must in my future ideas for apps i want to create with Budibase. Though i guess i'll most of the times set up self-hosted systems.

**Reply by mjashanks** [reply]
We think that the User Groups ("Workspaces") will be a paid feature on self-hosted as well. We won't be working on that for a while though, so can't tell you all that much about it, or when we will get to it.

The "no access" feature is being worked on this month and will be open-source.

**Reply by KernOil** [reply]
Thanks for the update @mjashanks I look forward to the "no access" feature!
